internal/manifest/iso_tree.go

```diff
@@ -74,80 +84,108 @@ func (p *ISOTree) getBuildPackages() []string {
 func (p *ISOTree) serialize() osbuild.Pipeline {
 	pipeline := p.Base.serialize()
 
-	kspath := "/osbuild.ks"
 	ostreeRepoPath := "/ostree/repo"
+	kernelOpts := fmt.Sprintf("inst.ks=hd:LABEL=%s:%s", p.isoLabel, p.KSPath)
 
-	pipeline.AddStage(osbuild.NewBootISOMonoStage(bootISOMonoStageOptions(p.anacondaPipeline.kernelVer,
-		p.anacondaPipeline.platform.GetArch().String(),
-		p.UEFIVendor,
-		p.anacondaPipeline.product,
-		p.anacondaPipeline.version,
-		p.isoLabel,
-		kspath),
-		osbuild.NewBootISOMonoStagePipelineTreeInputs(p.anacondaPipeline.Name())))
+	pipeline.AddStage(osbuild.NewMkdirStage(&osbuild.MkdirStageOptions{
+		Paths: []osbuild.Path{
+			{
+				Path: "images",
+			},
+			{
+				Path: "images/pxeboot",
+			},
+		},
+	}))
 
-	kickstartOptions, err := osbuild.NewKickstartStageOptions(kspath, "", p.Users, p.Groups, makeISORootPath(ostreeRepoPath), p.osTreeRef, p.OSName)
+	inputName := "tree"
+	copyStageOptions := &osbuild.CopyStageOptions{
+		Paths: []osbuild.CopyStagePath{
+			{
+				From: fmt.Sprintf("input://%s/boot/vmlinuz-%s", inputName, p.anacondaPipeline.kernelVer),
+				To:   "tree:///images/pxeboot/vmlinuz",
+			},
+			{
+				From: fmt.Sprintf("input://%s/boot/initramfs-%s.img", inputName, p.anacondaPipeline.kernelVer),
+				To:   "tree:///images/pxeboot/initrd.img",
+			},
+		},
+	}
+	copyStageInputs := osbuild.NewPipelineTreeInputs(inputName, p.anacondaPipeline.Name())
+	copyStage := osbuild.NewCopyStageSimple(copyStageOptions, copyStageInputs)
+	pipeline.AddStage(copyStage)
+
+	squashfsOptions := osbuild.SquashfsStageOptions{
+		Filename: "images/install.img",
+		Compression: osbuild.FSCompression{
+			Method: "lz4",
+		},
+	}
+	squashfsStage := osbuild.NewSquashfsStage(&squashfsOptions, p.rootfsPipeline.Name())
+	pipeline.AddStage(squashfsStage)
+
+	isoLinuxOptions := &osbuild.ISOLinuxStageOptions{
+		Product: osbuild.ISOLinuxProduct{
+			Name:    p.anacondaPipeline.product,
+			Version: p.anacondaPipeline.version,
+		},
+		Kernel: osbuild.ISOLinuxKernel{
+			Dir:  "/images/pxeboot",
+			Opts: []string{kernelOpts},
+		},
+	}
+	isoLinuxStage := osbuild.NewISOLinuxStage(isoLinuxOptions, p.anacondaPipeline.Name())
+	pipeline.AddStage(isoLinuxStage)
+
+	filename := "images/efiboot.img"
+	pipeline.AddStage(osbuild.NewTruncateStage(&osbuild.TruncateStageOptions{
+		Filename: filename,
+		Size:     "20MB",
```

Comment:
Can't we take this from the partition table?

Reply:
Yes.  My bad.  Thanks for catching it.

Reply:
Doing it in a follow up is fine.